operator/apis/platform/v1alpha1/securedcluster_types.go

```diff
@@ -340,6 +344,24 @@ type LocalScannerComponentSpec struct {
 	DB *DeploymentSpec `json:"db,omitempty"`
 }
 
+// LocalScannerV4ComponentSpec defines settings for the "scanner V4" component in SecuredClusters
+type LocalScannerV4ComponentSpec struct {
+	// If you do not want to deploy the Red Hat Advanced Cluster Security local Scanner V4, you can disable it here
+	// (not recommended).
+	// If you do so, all the settings in this section will have no effect.
+	//+kubebuilder:default=AutoSense
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Scanner Component",order=1
+	ScannerComponent *LocalScannerComponentPolicy `json:"scannerComponent,omitempty"`
+
+	// Settings pertaining to the indexer deployment.
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,order=2,xDescriptors={"urn:alm:descriptor:com.tectonic.ui:fieldDependency:.scannerComponent:AutoSense"}
```

Comment:
What does `fieldDependency:.scannerComponent:AutoSense` mean here?
I assume `scannerComponent` and also be set to something like `Enabled` -- would this have impact on this `Indexer` field?

Reply:
For LocalScannerComponentPolicy the enum has two possible values `AutoSense` or `Disabled`. As far as I understood this dependency hides the indexer config fields if the value for ScannerComponent is not AutoSense